packages/MainUI/app/window/[windowId]/[tabId]/[recordId]/page.tsx

```diff
@@ -4,9 +4,16 @@ import { useSingleDatasource } from '@workspaceui/etendohookbinder/src/hooks/use
 import DynamicFormView from '../../../../../screens/Form/DynamicFormView';
 import { useParams } from 'next/navigation';
 import { useMetadataContext } from '../../../../../hooks/useMetadataContext';
+import { Toolbar } from '../../../../../components/Toolbar';
+import { styles } from './styles';
 
 export default function Page() {
-  const { recordId } = useParams<{ recordId: string }>();
+  const { windowId, tabId, recordId } = useParams<{
+    windowId: string;
+    tabId: string;
+    recordId: string;
```

Comment:
I think we're repeating this everywhere. We should declare an interface and use it